test/index.js

```diff
@@ -1017,8 +1017,8 @@ test('extend', function() {
   eq(Z.extend.length, 2);
   eq(Z.extend.name, 'extend');
 
-  eq(Z.extend(length, []), [0]);
-  eq(Z.extend(length, [1, 2, 3]), [3]);
+  eq(Z.extend(length, []), []);
+  eq(Z.extend(length, [1, 2, 3]), [3, 2, 1]);
```

Comment:
Let's change this to something like `eq(Z.extend(ss => ss.join(''), ['x', 'y', 'z']), ['xyz', 'yz', 'z'])` to assert that the correct array slices are provided.